pkg/dockerregistry/server/auth.go

```diff
@@ -535,3 +545,11 @@ func verifyPruneAccess(ctx context.Context, client client.SubjectAccessReviews)
 	}
 	return nil
 }
+
+func isMetricsBearerToken(ctx context.Context, token string) bool {
+	config := ConfigurationFrom(ctx)
+	if config.Metrics.Enabled {
+		return config.Metrics.Secret == token
+	}
```

Comment:
Actually, does the registry support cert auth (is it using the correct filter for that)?  If not, I'm ok with static secret.